api/src/main/java/com/velocitypowered/api/event/connection/PreLoginEvent.java

```diff
@@ -46,4 +52,57 @@ public String toString() {
                 ", result=" + result +
                 '}';
     }
+    
+    /**
+     * Represents an "allowed/allowed with online mode/denied" result with a reason allowed for denial.
+     */
+    public static class PreLoginComponentResult extends ComponentResult {
+
+        private static final PreLoginComponentResult ALLOWED = new PreLoginComponentResult((Component) null);
+        private static final PreLoginComponentResult ALLOWED_ONLINEMODE = new PreLoginComponentResult(true);
+
+        private final boolean onlineMode;
+        /**
+         * Allows to enable a online mode for the player connection, when Velocity running in offline mode
+         * Does not have any sense if velocity running in onlineMode;
+         * @param onlineMode if true, offline uuid will be used for player connection if Velocity run in offlineMode
+         */
```

Comment:
Move this Javadoc to `#allowedOnlineMode()` (but see below for a further clarification to make).